### ISSUE TITLE
Criterion Confidence Codes vs. Evidence Strength Codes

### ISSUE THREAD

**Issue by larrybabb** [clingen-data-model/clingen-interpretation#52]
Here's a first stab at separating the criterion assessment outcome into confidence and strength codes to provide for the SEPIO harmonization that ties these assertions to the CriterionAssessment and Evidence Line, respectively.  Please review and comment.
---
@bpow , @cbizon 
It seems like we are on track to split the notion of "assessing" the outcome of 2 separate events: 

1. Criterion Confidence - when an agent wants to make an explicit statement regrading the evaluation of an individual ACMG rule (criterion) - aka Criterion Assessment:
  a. **Not met** - data exists but does not satisfy the rule
  b. **Insufficient** - not enough data is available to perform an evaluation
  c. **Not evaluated** - maybe the same as Insufficient, but could mean, I skipped this and didn't try to find the data yet.
  d. **Met** - data exists that does satisfy the rule (not a statement of the quality of data nor how this assessment should be applied as input to the variant interpretation evaluation process).
  e. **Refutes** - (does not apply to all rules, some rules have an inverse rule e.g. PS3-BS3) explicit evidence demonstrates that this rule is NOT met.  (Not sure why this is not "Not Met" unless we are saying this impacts the subsequent scoring input to variant interpretation.  If so, then should this impact the way the Evidence Strength is applied - see next )   

2. Evidence Strength - This is generically considered to be a set of values that would apply to any evidence (I think).  However, for our purposes, we would limit the choices to the ACMG values and use this only on evidence that is built on top of CriterionAssessments (see above).
  a. **Path Very Strong**
  b. **Path Strong**
  c. **Path Moderate**
  d. **Path Supporting**
  e. **Benign Supporting**
  f. **Benign Moderate**
  g. **Benign Strong**
  h. **Benign Stand Alone**
  i. **Inconclusive** (? adding to cover the use case that an agent needs to link criterion assessments and their related supporting evidence lines without impacting the scoring rules. (i.e. insufficient, not met, etc.. or just because they don't want it to be considered during interpretation calculations).

I was considering filtering out the "Path" and "Benign" qualifiers to reduce the set to only VS, S, M, P, SA. But, I think it is critical to keep the values specific.  It seems that the strength is more of an evaluation of how far along the continuum between Benign and Pathogenic a given CriterionAssessment evidence line is determined to be by an individual agent.  It does allow for the notion that a Refuting Criterion Assessment for what would be considered a Path Strong rule (PSx) could be given a strength of "Benign Moderate" or something like that.  Not sure if that is reasonable or useful or logical, so please verify.

**Comment by larrybabb** [comment]
I'm also advocating for calling the "confidence" assessment "CriterionEvaluation" since it deals with assessing not only whether a given rule/criterion was evaluated, but also what the general evaluation of the available supporting evidence is in the context of the rule/criterion.

**Comment by bpow** [comment]
For the Criterion Confidence, I'm mostly in agreement. I'm not entirely sure of the difference of "not met" and "insufficient", though... could these be combined to one term?

For the Evidence Strength, this gets back to the discussion of considering magnitude and direction as two different variables ("{Benign, Supporting}") vs. as a combined attribute ("Benign Supporting").

The stickler in me thinks that maybe we should just represent the strength in the EvidenceLine (Inconclusive, Supporting, Moderate, Strong, Stand Alone), since that tells how far the EvidenceLine "moves the needle", and any given criterion can only move the needle in one direction or the other. Maybe one more "strength" of "Noncontributory" that we might use for evaluated but refuted criteria?

**Comment by larrybabb** [comment]
@bpow @cbizon We should try to resolve this soon based on the view that Matt Brush provided. There's both structural changes (attributes on evidenceLines and assertions  - I think) as well as specifically what codes/terms we want to use.

**Comment by cbizon** [comment]
tl;dr: I agree with Larry's proposal.

I agree with having the Assessment having its own result which is something like (met, not met, others), and the evidence line having the strength in it, as defined by the ACMG terms.

Both of these term lists are going to be ACMG specific, because the scoring model we are trying to describe is specialized.  

So the only question, I think is the terms for the Assessment.

In terms of implementing the bare minimum for ACMG, we would only need two values:
* Yes, use this as a positive rule in calculating the pathogenicity (met)
* No, do not use this as a positive rule in calculating the pathogenicity (not met)

Any other values that we want to add are purely to increase the expressivity of the model.  That is, we are just adding information saying why we consider the rule to be met or not met.  In terms of creating the overall interpretation, these terms will be binarized.

Larry's proposal above includes one term for "met" and 4 for "not met".  I interpret them as follows:
* Not evaluated: not met because we didn't try
* Insufficient evidence: Not enough evidence could be collected to decide whether the rule could be met
* Not met: enough evidence was collected, but the rule was not satisified.
* Refuted: Lots of evidence that the rule was not met, and almost no amount of future evidence will ever convince me otherwise.

Refuted is probably the easiest to drop (combine with not met).  The goal of it was to mark assessments that would not need further review, and I guess I still think that's useful, so I would keep it.

Sorry for working out my thoughts via text, but I think I've come down on accepting this set of terms.

**Comment by cbizon** [comment]
But I don't like the term "CriterionConfidence"  I think I prefer AssessmentResult or something.  It's not the confidence in the result - it IS the result.
